pkg/controller/common/reconciler/secret.go

```diff
@@ -227,3 +228,8 @@ func GarbageCollectAllSoftOwnedOrphanSecrets(ctx context.Context, c k8s.Client,
 	}
 	return nil
 }
+
+// restrictedToOwnerNamespace returns true if a resource should have its owner in the same namespace, based on the kind of owner.
+func restrictedToOwnerNamespace(kind string) bool {
+	return kind != policyv1alpha1.Kind
```

Comment:
Does it no depend on where the policy lives? If it is not in the operator namespace then the same namespace restriction should still apply? But I guess it does not matter much given he additional list options that restrict to a specific owner. I would probably still at a comment explaining the special case

Reply:
No, the file settings Secret can be in the Elasticsearch (owner) namespace which can be different from the policy (softowner) namespace.

`restrictedToSoftOwnerNamespace` would be more precise. Here we want to be able to remove the file settings Secret where the soft owner no longer exists and we have missed the clean reset of the Secret when deleting the soft owner.